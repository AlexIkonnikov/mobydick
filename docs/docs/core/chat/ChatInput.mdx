import ChatInput from './assets/ChatInput.png';

<img src={ChatInput} />

Reusable ChatInput component with predefined styles for chat


## Example

```tsx
import {useState} from 'react';

import {
  ChatInput,
} from '@lad-tech/mobydick-core';

export const ChatWidget = () => {
  const [input, setInput] = useState('');
  const [myMessages, setMyMessages] = useState([
    {time: '12-12-1998', isMe: true, message: 'message'},
  ]);

  const onSend = () => {
    setMyMessages([
      ...myMessages,
      {
        time: new Date().toString(),
        isMe: true,
        message: input,
      },
    ]);
    setInput('');
  };

  return (
    <ChatInput>
      <ChatInput.ChatInputField value={input} onChangeText={setInput} />
      <ChatInput.ChatPressableIcon name={'icon-send'} onPress={onSend} />
    </ChatInput>
  );
};

```

## `Props`

### [`View Props`](https://reactnative.dev/docs/view#props)

Inherits [View Props](https://reactnative.dev/docs/view#props).

### `containerStyle`

| TYPE                                                             |
|:-----------------------------------------------------------------|
| [ViewStyle](https://reactnative.dev/docs/view-style-props#props) |

Custom styles for container


## `Componetns`

### `ChatInputField`

See [ChatInputField](./ChatInputField.md)

### `ChatInputField`

See [ChatInputField](./ChatPressableIcon.md)

