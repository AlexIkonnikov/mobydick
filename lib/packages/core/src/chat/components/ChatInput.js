"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_native_1 = require("react-native");
const react_native_safe_area_context_1 = require("react-native-safe-area-context");
const View_1 = __importDefault(require("../../basic/components/View/View"));
const styles_1 = require("../../styles");
const useStyles_1 = __importDefault(require("../../styles/hooks/useStyles"));
const ChatInputField_1 = __importDefault(require("./ChatInputField"));
const ChatPressableIcon_1 = __importDefault(require("./ChatPressableIcon"));
const ChatInput = props => {
    const { children, style, enabled = true, containerStyle } = props;
    const [styles] = (0, useStyles_1.default)(stylesCreate);
    const insets = (0, react_native_safe_area_context_1.useSafeAreaInsets)();
    return (<react_native_1.KeyboardAvoidingView style={[
            {
                marginBottom: insets.bottom,
            },
            style,
        ]} behavior={react_native_1.Platform.OS === 'ios' ? 'padding' : 'height'} enabled={enabled} keyboardVerticalOffset={0}>
      <View_1.default style={[styles.container, containerStyle]}>{children}</View_1.default>
    </react_native_1.KeyboardAvoidingView>);
};
const stylesCreate = (0, styles_1.createStyles)(({ spaces, colors }) => ({
    container: {
        flexDirection: 'row',
        alignItems: 'flex-end',
        backgroundColor: colors.BgPrimary,
        width: '100%',
        paddingHorizontal: spaces.Space20,
        paddingVertical: spaces.Space16,
    },
}));
ChatInput.ChatInputField = ChatInputField_1.default;
ChatInput.ChatPressableIcon = ChatPressableIcon_1.default;
exports.default = ChatInput;
