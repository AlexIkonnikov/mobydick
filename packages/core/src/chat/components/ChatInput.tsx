import React, {FC, PropsWithChildren} from 'react';
import {StyleSheet} from 'react-native';

import View from '../../basic/components/View/View';
import useStyles from '../../styles/theme/hooks/useStyles';
import {IThemeContext} from '../../styles';

import ChatInputField from './ChatInputField';
import ChatPressableIcon from './ChatPressableIcon';

const ChatInput: FC<PropsWithChildren> & {
  ChatInputField: typeof ChatInputField;
  ChatPressableIcon: typeof ChatPressableIcon;
} = props => {
  const {children} = props;
  const [styles] = useStyles(stylesCreate);

  return <View style={styles.container}>{children}</View>;
};

const stylesCreate = ({colors, spaces}: IThemeContext) =>
  StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'flex-end',
      backgroundColor: colors.BgPrimary,
      width: '100%',
      paddingHorizontal: spaces.Space20,
      paddingVertical: spaces.Space16,
    },
  });

ChatInput.ChatInputField = ChatInputField;
ChatInput.ChatPressableIcon = ChatPressableIcon;
export default ChatInput;
