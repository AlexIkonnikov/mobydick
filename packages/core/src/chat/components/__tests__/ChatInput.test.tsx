import {render} from '@testing-library/react-native';
import React from 'react';

import ChatInput from '../ChatInput';

describe('@npm/mobydick-core/ChatInput', () => {
  it('renders correctly', () => {
    const {toJSON} = render(
      <ChatInput>
        <ChatInput.ChatInputField value={'value'} />
        <ChatInput.ChatPressableIcon
          name={'icon-location'}
          onPress={() => null}
        />
      </ChatInput>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
