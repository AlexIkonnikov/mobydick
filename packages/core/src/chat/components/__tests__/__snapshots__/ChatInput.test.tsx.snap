// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-core/ChatInput renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "flex-end",
      "backgroundColor": "#FFF",
      "flexDirection": "row",
      "paddingHorizontal": 30,
      "paddingVertical": 24,
      "width": "100%",
    }
  }
>
  <View
    style={
      [
        {
          "minWidth": 195,
        },
        [
          {
            "flex": 1,
            "flexDirection": "column",
            "minWidth": undefined,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "flex-start",
            "backgroundColor": "#EEEFF1",
            "borderColor": "transparent",
            "borderRadius": 12,
            "borderWidth": 1.5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 12,
            "paddingLeft": 27,
            "paddingRight": 24,
            "paddingVertical": 18,
          },
          {
            "minHeight": 120,
          },
          [
            {
              "alignItems": "center",
              "borderWidth": undefined,
              "marginRight": 3,
              "marginVertical": 0,
              "maxHeight": 294,
              "minHeight": 57,
              "minWidth": 367.5,
              "paddingVertical": 12,
              "width": "100%",
            },
            undefined,
          ],
        ]
      }
    >
      <TextInput
        accessibilityLabel="chatInputField"
        allowFontScaling={true}
        editable={true}
        multiline={true}
        numberOfLines={1}
        onBlur={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#9BA1B0"
        selectionColor="#2B78EE"
        style={
          [
            {
              "flex": 1,
              "padding": 0,
            },
            {
              "color": "#20242D",
              "fontFamily": "Inter-Regular",
              "fontSize": 24,
              "lineHeight": 30,
            },
            [
              {
                "paddingTop": 0,
              },
              undefined,
            ],
          ]
        }
        testID="TextInput"
        value="value"
      />
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E5F0FF",
        "borderRadius": 12,
        "height": 57,
        "justifyContent": "center",
        "marginLeft": 9,
        "opacity": 1,
        "width": 57,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        {
          "color": "#2B78EE",
          "fontFamily": "Neotis",
          "fontSize": 36,
        }
      }
    >
      î¨­
    </Text>
  </View>
</View>
`;
